src/output.rs

```diff
@@ -54,6 +54,43 @@ fn print_trailing_slash<W: Write>(
     Ok(())
 }
 
+// Trying to copy: https://www.gnu.org/software/coreutils/quotes.html
+fn path_needs_quoting(path: &str) -> i8 {
+    // If it contains any special chars we return single quote
+    if path.contains(" ") || path.contains("$") || path.contains("\"") {
+        return 1;
```

Comment:
It would be better to use an enumeration instead of an `i8`, where it isn't obvious what the different numbers mean.